x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -287,6 +282,19 @@ func runContainerCmd(streams *cli.IOStreams, cmd *cobra.Command, cfg setupConfig
 		}
 	}
 	if cfg.Fleet.Enroll {
+		if cfg.FleetServer.Enable {
+			if client == nil {
+				client, err = kibanaClient(cfg.Kibana, cfg.Kibana.Headers)
+				if err != nil {
+					return err
+				}
+			}
+			err = ensureServiceToken(streams, client, &cfg)
```

Comment:
Shouldn't this move to the line 273? That way the call to `kibanaSetup` uses a `service_token` as well